### DISCUSSION TITLE
airodump-ng is no longer working in Qualcomm devices

### DISCUSSION THREAD

**Discussion by spiral009** [aircrack-ng/aircrack-ng#2555] [General]
I used to to enter monitor mode and use the airodump-ng just fine but after I switch the custom rom I have issues 
I use these commands to switch between monitor mode and managed mode
```
echo 4 > /sys/module/wlan/parameters/con_mode
echo 0 > /sys/module/wlan/parameters/con_mode
```
Now if I run these commands after I turn off the wifi I can turn it back on and connect to APs just fine but if I were to switch to monitor mode and use `airodump-ng wlan0`
The tool will run for only few seconds then freezes 
![Screenshot_٢٠٢٣٠٧١٣-١٣٤٠٥٢_الاستديو](https://github.com/aircrack-ng/aircrack-ng/assets/39057208/2f5ba96b-b9f4-4cc7-820d-12694578f686)
If I were to run `ps` command I see two instance of airodump-ng running at the same time 
![Screenshot_٢٠٢٣٠٧١٣-١٣٢٥٤٩_Termux](https://github.com/aircrack-ng/aircrack-ng/assets/39057208/dd7d362b-57e2-4e5a-92ab-249ee1c6bb94)
And once I quit out of airodump-ng using ctr+C or Z airodump-ng keeps on running 
![Screenshot_٢٠٢٣٠٧١٣-١٣٢٥٥١_Termux](https://github.com/aircrack-ng/aircrack-ng/assets/39057208/74bb0fa0-98b8-4c55-9106-239dbd177bac)
And after using airodump-ng if I switch the interface back to managed mode the WiFi won't work if I turn it on unless I reboot the device.
I didn't have any of these issues with the previous rom so I don't know if it's the rom or a new version of airodump-ng iamt not aware of ? all I know is the kernel is the Same  I built it myself.
My guess is something constantly using wlan0 but I can't find wlan0 in /dev to run `lsof /dev/wlan0` and see what's using this interface 
[airodump-ng.strace.txt](https://github.com/aircrack-ng/aircrack-ng/files/12071920/airodump-ng.strace.txt)

**Comment by spiral009** [comment]
Update: i can use airodump-ng on a specific channel but not multiple if i run with -c 1,2,3 the tool Will freeze and cause troubles 
And it's looking on channel zero which is weird

**Comment by ZerBea** [comment]
After some research my suspect is the driver:
https://github.com/ZerBea/hcxdumptool/issues/332

Looks like aireplay-ng injection test failed, too (due to driver problems):
https://github.com/ZerBea/hcxdumptool/issues/332#issuecomment-1650337261
